src/ui/widget.cpp

```diff
@@ -199,14 +199,23 @@ void Widget::setTheme(Theme* theme)
 void Widget::setStyle(Style* style)
 {
   assert_ui_thread();
-
-  m_style = style;
-  m_border = m_theme->calcBorder(this, style);
-  m_bgColor = m_theme->calcBgColor(this, style);
-  m_minSize = m_theme->calcMinSize(this, style);
-  m_maxSize = m_theme->calcMaxSize(this, style);
-  if (style->font())
-    m_font = AddRef(style->font());
+  if (style) {
```

Comment:
Same as in the grid case, I think this should never happen, we could do:
```
  ASSERT(style);
  if (!style)
    style = Theme::getDefaultStyle();
  ...
```
The default style should already have some default style constructed in some way so `m_theme->calc...` calls return the default values.